stock_picking_invoice_link/models/stock_picking.py

```diff
@@ -22,12 +25,18 @@ def action_view_invoice(self):
         """
         self.ensure_one()
         form_view_name = "account.view_move_form"
-        action = self.env.ref("account.action_move_out_invoice_type").sudo()
-        result = action.read()[0]
+        result = self.env["ir.actions.act_window"]._for_xml_id(
+            "account.action_move_out_invoice_type"
+        )
         if len(self.invoice_ids) > 1:
             result["domain"] = "[('id', 'in', %s)]" % self.invoice_ids.ids
         else:
             form_view = self.env.ref(form_view_name)
             result["views"] = [(form_view.id, "form")]
             result["res_id"] = self.invoice_ids.id
         return result
+
+    @api.depends("invoice_ids")
+    def _compute_invoice_count(self):
+        for order in self:
+            order.invoice_count = len(order.invoice_ids)
```

Comment:
It's better to use `read_group` for performance reasons.

Reply:
It's a M2M field, how you suggest to do it?

Reply:
Ah, OK, I see...